### PR TITLE
Feat(org-settings): finalize identity/branding controls, authz, and kiosk branding consistency

### PR DESCRIPTION
**Pull request by yuvraj-shishodia**
## Summary
  This PR includes only the finalised scope from the org-settings workstream.
  It is intentionally larger because the final behaviour required coordinated frontend + backend updates.

  ## Finalised Changes Included

https://github.com/user-attachments/assets/82522e9a-e9fc-41a9-b9b2-b680f09ae0fc

https://github.com/user-attachments/assets/73093d91-537b-43cb-aef4-6864dba2cfe7

https://github.com/user-attachments/assets/3b6fc5f9-e370-4d1c-8dd9-809701e7eaf5

  ### 1) Admin org context and switcher stability
  - Improved organisation switcher UX and search behaviour.
  - Synced selected organisation name in admin headers/layout without requiring manual refresh.
  - Ensured organisation context remains consistent across admin screens.

  ### 2) Organisation Settings UX and save model
  - Split settings actions into:
    - Save Identity
    - Save Branding
  - Improved image adjustment flow for organisation logo / idle screensaver.
  - Added reset/remove controls where finalised.

  ### 3) Kiosk branding consistency
  - Propagated organisation accent colour across kiosk journey:
    - campaign list
    - campaign details
    - payment
    - email confirmation
    - loading states
  - Reduced default-colour flashes and aligned loading visuals with org branding.

  ### 4) Permission model hardening (final)
  - Added new permissions:
    - `change_org_identity`
    - `change_org_branding`
  - Granted these to `admin` and `super_admin` defaults.
  - Frontend now gates identity vs branding controls/actions by these permissions.
  - Backend enforces section-level authorisation and prevents cross-section writes.
  - User management backend prevents assigning these org-settings permissions to non-admin/non-super-admin roles.

  ### 5) Backend validation updates
  - Organisation display name uniqueness check during settings update.
  - Section-aware payload handling for organisation settings updates.

  ### 6) Legacy lint/type cleanup (no lint exception)
  - Removed file-specific lint exception approach.
  - Fixed legacy lint/type issues in `AdminDashboard` so strict linting passes without per-file rule bypass.

  ## Review Focus
  Please review carefully for:
  - Section-level permission enforcement (`identity` vs `branding`) end-to-end.
  - Org name/switch sync behaviour across admin pages.
  - Kiosk accent-colour propagation and loading-state consistency.
  - Backend validation + authorisation paths in organisation settings handlers.
  - No regression in existing admin role flows.

  ## Validation
  - TypeScript checks passed locally (`tsc --noEmit`).
  - Targeted ESLint checks passed for modified areas.
  - Backend handler syntax checks passed.

  ## Risk / Impact
  - Medium-to-high surface area (frontend + backend).
  - Low architectural risk; changes are mostly enforcement, propagation, and correctness fixes.
  - Primary risk is behavioural regression in edge permission flows; needs careful QA.

reviewers_note: |
  This PR is expected to take longer than usual to review.
  Please prefer a deep functional review over a quick skim due to cross-cutting impact.

### CODE REVIEW

**Comment by yuvraj-shishodia** [comment]
I will backfill the new user permissions(organisation settings related) for existing users.

**Review by dp-singh — COMMENTED** [review]
Reviewed the org-settings/authz rollout line by line. I found three blocking issues in the new permission/rules path.

**Comment by dp-singh** [comment]
Follow-up from the deeper pass: I do not see any PR-added tests for the new org-settings/authz surface, and I also do not see existing repo tests covering these paths. The PR changes 69 files, but `gh pr diff --name-only` shows no `*.test.*` / `*.spec.*` files in the submitted diff.

The green coverage check therefore does not demonstrate changed-path coverage here. At minimum I would want examples around:
- `updateOrganizationSettings` rejecting duplicate display names and off-bucket asset references
- Firestore/rules behavior for direct `organizations/{orgId}.settings` writes versus the backend function path
- org admins adding campaign tags after the `settings`-only organization update rule change
- permission separation proving an admin without `change_org_identity` cannot still edit identity fields

Without coverage like that, the highest-risk regressions in this PR are still unproven.

**Comment by dp-singh** [comment]
One more coverage-specific note after fetching `origin/main` and checking the repo/workflow definitions directly: the PR's `Code Quality & Coverage` check is not evidence that coverage was actually measured here.

`.github/workflows/code-quality.yml` only runs coverage when `package.json` contains a `test:coverage` script, but this repo only defines `test:run` (`vitest run --coverage`). In the current workflow, the coverage step falls through to `echo "⚠️ No test:coverage script configured"` and can still report success.

So for this PR:
- there are no PR-added test/spec files in the diff
- the green `Coverage` status does not prove changed-path coverage
- the current workflow setup can produce a false sense of safety for exactly this kind of large authz/rules/UI change

Relevant files: `.github/workflows/code-quality.yml`, `package.json`.

File: backend/firestore.rules [hunk 1]
```diff
@@ -133,9 +170,17 @@ service cloud.firestore {
...
+
+      // Only privileged admins can create/delete organization records
+      allow create, delete: if isPrivilegedAdmin();
+
+      // Organization admins can only update the settings field for their own org.
+      // Privileged admins can update any field.
+      allow update: if isPrivilegedAdmin() ||
+                      (
+                        isOrganizationSettingsAdmin(orgId) &&
+                        request.resource.data.diff(resource.data).affectedKeys().hasOnly(['settings'])
```
[the remark this answers is not in the document]
Comment (yuvraj-shishodia) [reply backend/firestore.rules]: Fixed. I updated the rule so org-local admins can update only settings and tags (no other org fields), which restores the existing tag-management flow while keeping the authz boundary tight.

File: backend/functions/handlers/organizationSettings.js [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+    hasOrgSettingsWriteAccessForPermission(callerData, IDENTITY_PERMISSION) ||
+    hasOrgSettingsWriteAccessForPermission(callerData, BRANDING_PERMISSION)
+  );
+};
+
+const hasOrgSettingsWriteAccessForPermission = (callerData, permission) => {
+  const role = typeof callerData?.role === 'string' ? callerData.role : '';
+  const permissions = Array.isArray(callerData?.permissions) ? callerData.permissions : [];
+
+  return (
```
[the remark this answers is not in the document]
Comment (yuvraj-shishodia) [reply backend/functions/handlers/organizationSettings.js]: Good catch. I removed the role-based bypass in hasOrgSettingsWriteAccessForPermission, so identity/branding access is now truly permission-driven. Admin/super_admin users must explicitly have change_org_identity and/or change_org_branding (or system_admin) for the corresponding changes. This makes the split permission model materially enforceable.

**Comment by yuvraj-shishodia** [comment]
> Follow-up from the deeper pass: I do not see any PR-added tests for the new org-settings/authz surface, and I also do not see existing repo tests covering these paths. The PR changes 69 files, but `gh pr diff --name-only` shows no `*.test.*` / `*.spec.*` files in the submitted diff.
> 
> The green coverage check therefore does not demonstrate changed-path coverage here. At minimum I would want examples around:
> 
> * `updateOrganizationSettings` rejecting duplicate display names and off-bucket asset references
> * Firestore/rules behavior for direct `organizations/{orgId}.settings` writes versus the backend function path
> * org admins adding campaign tags after the `settings`-only organization update rule change
> * permission separation proving an admin without `change_org_identity` cannot still edit identity fields
> 
> Without coverage like that, the highest-risk regressions in this PR are still unproven.

Implemented coverage for the new org-settings/authz handler surface in backend/functions/handlers/organizationSettings.test.js. Added tests for duplicate display-name rejection, off-bucket asset rejection, and split permission enforcement (identity vs branding), including a positive branding-only path. I also extended the Firebase admin test mock to support collection scans used by uniqueness checks.
For Firestore rules behaviour (direct organisations/{orgId}.settings writes vs function path), I agree this should be covered with emulator rules tests; I can add that as a follow-up test harness task if we want strict changed-path coverage at rules-engine level.

**Comment by yuvraj-shishodia** [comment]
> One more coverage-specific note after fetching `origin/main` and checking the repo/workflow definitions directly: the PR's `Code Quality & Coverage` check is not evidence that coverage was actually measured here.
> 
> `.github/workflows/code-quality.yml` only runs coverage when `package.json` contains a `test:coverage` script, but this repo only defines `test:run` (`vitest run --coverage`). In the current workflow, the coverage step falls through to `echo "⚠️ No test:coverage script configured"` and can still report success.
> 
> So for this PR:
> 
> * there are no PR-added test/spec files in the diff
> * the green `Coverage` status does not prove changed-path coverage
> * the current workflow setup can produce a false sense of safety for exactly this kind of large authz/rules/UI change
> 
> Relevant files: `.github/workflows/code-quality.yml`, `package.json`.

Good catch. I fixed this by adding an explicit test: coverage script in package.json and updating the Code Quality workflow to fail hard when coverage is missing/failing (instead of echoing and continuing). I also updated the report’s overall pass criteria to include coverage outcome. So a green Coverage check now reflects actual coverage execution.
@dp-singh

**Comment by dp-singh** [comment]
Re-reviewed the delta from `ee81feaad1f7eb9cd7948fac5d246dd3fe18db21` to the current head `7eb99708b3c34227ef9a8e04882dcb6a33612a12`. The original authz/rules blockers look addressed in code, but I still see one remaining testing blocker:

The new handler coverage in `backend/functions/handlers/organizationSettings.test.js` does not appear to be part of the executed CI test set. Root CI still runs `npm run test:run` from `package.json`, which is `vitest run --coverage`, and `vitest.config.ts` still limits discovery to `include: ["src/**/*.test.ts"]`. Separately, `backend/functions/package.json` defines its own `jest --runInBand` script, but none of the current workflows invoke that functions test command.

So the PR now contains the right tests on disk, but they still look inert from CI's point of view. I would not treat those new org-settings handler tests as proven until the workflow actually runs them (or Vitest discovery is expanded to include them, if that's the intended runner).

Relevant files: `vitest.config.ts`, `package.json`, `backend/functions/package.json`, `backend/functions/handlers/organizationSettings.test.js`.

**Comment by yuvraj-shishodia** [comment]
> Re-reviewed the delta from `ee81feaad1f7eb9cd7948fac5d246dd3fe18db21` to the current head `7eb99708b3c34227ef9a8e04882dcb6a33612a12`. The original authz/rules blockers look addressed in code, but I still see one remaining testing blocker:
> 
> The new handler coverage in `backend/functions/handlers/organizationSettings.test.js` does not appear to be part of the executed CI test set. Root CI still runs `npm run test:run` from `package.json`, which is `vitest run --coverage`, and `vitest.config.ts` still limits discovery to `include: ["src/**/*.test.ts"]`. Separately, `backend/functions/package.json` defines its own `jest --runInBand` script, but none of the current workflows invoke that functions test command.
> 
> So the PR now contains the right tests on disk, but they still look inert from CI's point of view. I would not treat those new org-settings handler tests as proven until the workflow actually runs them (or Vitest discovery is expanded to include them, if that's the intended runner).
> 
> Relevant files: `vitest.config.ts`, `package.json`, `backend/functions/package.json`, `backend/functions/handlers/organizationSettings.test.js`.

 You’re right that the new backend/functions/handlers/organizationSettings.test.js was not part of the executed CI path before. I fixed this by wiring backend/functions Jest tests into CI workflows directly:
 
 1. ci.yml now installs backend/functions dependencies and runs functions tests, and
 
 2. code-quality.yml now runs/report functions tests as a separate check.
I also added a root test: functions script to make this explicit and reusable.
So the org-settings handler tests are now part of executed CI, not just present on disk.

**Comment by yuvraj-shishodia** [comment]
I updated the 2 failing assertions to match current sanitiser behaviour (leading apostrophe is prepended at the start of the risky field).

Validated:

npm run test:functions -- --runInBand services/giftAidExport.test.js handlers/organizationSettings.test.js passes.

**Comment by CodrJitesh** [comment]
@yuvraj-shishodia I can see in the attached video that when setting a name for the org, an error appeared saying "organization display name already exists." What data is this check referencing?
The organization name is already unique, so requiring the display name to also be unique doesn’t make sense.

**Comment by CodrJitesh** [comment]
@yuvraj-shishodia Seems like the image streches only vertically when zooming in for cropping it, please make sure to make it consistent.

**Comment by yuvraj-shishodia** [comment]
> @yuvraj-shishodia I can see in the attached video that when setting a name for the org, an error appeared saying "organization display name already exists." What data is this check referencing? The organization name is already unique, so requiring the display name to also be unique doesn’t make sense.

Good catch. @CodrJitesh 
That check currently scans the entire organisation's collection and compares normalised names against:

settings.displayName (if present), otherwise
name / organizationName fallback
So right now, displayName is enforced as globally unique across orgs (case/spacing-insensitive), not just organizationId.
You’re right, this is stricter than needed if only the org identity (organizationId) must be unique. I can remove/relax this global display-name uniqueness check in the next patch, but I think we do want organisations' names to be unique.

**Comment by yuvraj-shishodia** [comment]
> @yuvraj-shishodia Seems like the image streches only vertically when zooming in for cropping it, please make sure to make it consistent.

@CodrJitesh 
Zoom is already uniform on both axes:

displayWidth = naturalWidth * baseScale * zoom
displayHeight = naturalHeight * baseScale * zoom

**Comment by CodrJitesh** [comment]
> > @yuvraj-shishodia I can see in the attached video that when setting a name for the org, an error appeared saying "organization display name already exists." What data is this check referencing? The organization name is already unique, so requiring the display name to also be unique doesn’t make sense.
> 
> Good catch. @CodrJitesh That check currently scans the entire organisation's collection and compares normalised names against:
> 
> settings.displayName (if present), otherwise name / organizationName fallback So right now, displayName is enforced as globally unique across orgs (case/spacing-insensitive), not just organizationId. You’re right, this is stricter than needed if only the org identity (organizationId) must be unique. I can remove/relax this global display-name uniqueness check in the next patch, but I think we do want organisations' names to be unique.

Yes, the organization names itself shall still remain unique, just remove the restriction on the display name.

**Comment by yuvraj-shishodia** [comment]
@CodrJitesh ,
Implemented. I removed the global display-name uniqueness check from updateOrganizationSettings, so settings.displayName is no longer enforced as unique across organisations. We now allow duplicate display names and continue to rely on organizationId as the unique identifier.
I also updated handler tests accordingly (allows duplicate display names across organisations) and verified the functions test passes.

**Comment by yuvraj-shishodia** [comment]
@CodrJitesh ,
Fixed. I changed the crop dialogue to true uniform zoom (single scale(zoom) transform with centre origin), and updated both pan bounds and crop-export math to use the same zoomed dimensions.

## before
https://github.com/user-attachments/assets/52a1ac16-a99a-4f6a-9202-d1024352ac85

## after
https://github.com/user-attachments/assets/12876b8d-95c2-4b40-a8e6-df7489d9b404


So now:

X and Y always use the same scale factor.
No independent horizontal/vertical stretching path remains.
Pan and crop output stay aligned with what you see in preview.
This addresses the one-axis-stretch behaviour you reported.

**Review by CodrJitesh — APPROVED** [review]
